lib/core/src/persist/sync.rs

```diff
@@ -259,4 +275,196 @@ impl Persister {
 
         Ok(())
     }
+
+    fn check_commit_update(con: &Connection, record_id: &str, last_updated_at: u32) -> Result<()> {
+        let mut stmt = Self::get_sync_outgoing_details_stmt(con)?;
+        let mut rows = stmt.query(named_params! {
+            ":id": record_id,
+        })?;
+
+        if let Some(row) = rows.next()? {
+            let sync_outgoing_details = Self::sql_row_to_sync_outgoing_details(row)?;
+
+            if sync_outgoing_details.updated_at > last_updated_at {
+                return Err(anyhow::anyhow!("Record has been updated while pulling"));
+            }
+        }
+
+        Ok(())
+    }
+
+    pub(crate) fn commit_receive_swap(
+        &self,
+        data: &ReceiveSyncData,
+        sync_state: SyncState,
+        last_updated_at: Option<u32>,
+    ) -> Result<()> {
+        let mut con = self.get_connection()?;
+        let tx = con.transaction_with_behavior(TransactionBehavior::Immediate)?;
+
+        if let Some(last_updated_at) = last_updated_at {
+            Self::check_commit_update(&tx, &sync_state.record_id, last_updated_at)?;
+        }
+
+        tx.execute(
+            "
+            UPDATE receive_swaps
```

Comment:
We always use Update statement? What if the record doesn't exist? Same for all other swaps